fairscale/nn/data_parallel/fully_sharded_data_parallel.py

```diff
@@ -438,6 +439,28 @@ def no_sync(self) -> Generator:
             for m, old_flag in old_flags:
                 m.require_backward_grad_sync = old_flag
 
+    @contextlib.contextmanager
+    def summon_full_params(self) -> Generator:
+        """
+        A context manager to expose full params for the underlying model.
+        Can be useful after forward/backward for a model to get the params
+        for additional processing or checking.
+
+        This can be used on inner FSDPs.
+        """
```

Comment:
perhaps add a note that this can't be used in the middle of a forward/backward pass, nor can a forward/backward pass be started from within the context manager

Reply:
good idea. will do.